src/Support/Diagnostic.cpp

```diff
@@ -30,7 +32,24 @@ mlir::LogicalResult Diagnostic::attributeOutOfRange(mlir::Operation &op,
                                     .concat(", ")
                                     .concat(std::to_string(validRange.max))
                                     .concat("]"));
-};
+}
+
+LogicalResult Diagnostic::operandHasUnexpectedRank(Operation &op,
```

Comment:
I would also change the name to `emitErrorOperandHasUnexpectedRank`

Reply:
yes will do